source/frontend-docs.rst

```diff
@@ -0,0 +1,218 @@
+setup, it takes very little additional work to ensure that JSDoc can read
+your code. You will need the ``jsdoc-babel`` plugin::
+
+    npm install --save-dev jsdoc-babel
+
+And you will need to add lines to your config file to add the plugin and to
+get JSDoc to recognize ``.jsx`` files as well as ``.js`` ones::
+
+    {
+      "plugins": ["node_modules/jsdoc-babel"],
+      "source": {
+        "includePattern": ".+\\.jsx?$",
+        /* ... */
+      },
+      /* ... */
+    }
+
+That's it. Assuming Babel is correctly configured to process your JSX code,
+JSDoc will now be able to handle it, too.
+
+Once the configuration file is set up, you can generate a JSDoc site from your
+JS by running this command (assuming ``./node_modules/.bin`` is in your
+``$PATH``)::
+
+    jsdoc -c .jsdoc.json
+
+That's it! Your JSDoc documentation will have been deposited in ``./docs/js``.
+
+JSDoc is not a library and doesn't expose a full-featured API. For that
+reason, to use JSDoc from within your build process, it should be treated as
+just another command line program to be run using Node's ``child_process``.
+
+JSDoc: suggested comment style
+==============================
+
+JSDoc offers a very rich selection of tags and other features and a great deal
+of flexibility in how it can be used. We recommend adhering to at least these
+conventions:
+
+* Create a doc comment for every module, class, function / method, and event.
+* For every function / method doc comment, include:
+    * A prose description of the function's purpose
+    * ``@param`` tags each expected parameter, including:
+        * name
+        * type
+        * prose description
+    * For Object-type parameters, include separate ``@param`` entries for each
+      property expected on the Object
+    * ``@returns`` tag giving the function's return value, including:
+        * type
+        * prose description
+    * ``@fires`` tags indicating events fired within the function body
+* If a function is a callback:
+    * Declare its doc comment with ``@callback``
+    * Specify the event(s) it's attached to with ``@listens``
+* Include ``@see`` references wherever another part of the code provides useful
+  context for interpreting the comment.
+
+For the full details of what JSDoc is capable of, see `the JSDoc website <http://usejsdoc.org/>`_.
+
+Examples
+========
+
+Comment style
+-------------
+
+An ES6 module can be documented by adding a JSDoc comment to its
+constructor. This comment includes a brief prose description of
+the class's nature; it explicitly identifies the function as a
+constructor using ``@constructs``; it spells out the type of its single
+required parameter, ``options``, and the attribute ``id`` that it
+requires; and it uses ``@see`` to link to the docs for the method
+``appSetup`` which does the constructor's heavy lifting::
+
+    export default class ShowMatches extends Model {
+      /**
+       * Create a new app state model.
```

Comment:
Is this the documentation for the class or the constructor?  The prose description reads like it's just documenting the constructor, but if JSDoc is going to use this comment for the whole module, maybe it should be broadened.